src/events/SDL_events.c

```diff
@@ -995,7 +996,11 @@ SDL_WaitEventTimeout(SDL_Event * event, int timeout)
         break;
     default:
         if (event && event->type == SDL_POLLSENTINEL) {
-            /* Reached the end of a poll cycle, and no timeout */
+            /* Reached the end of a poll cycle, and not willing to wait */
+            if (SDL_AtomicGet(&SDL_sentinel_pending) > 0) {
```

Comment:
Would this be better inside the check in `SDL_PeepEventsInternal`? (i'm not sure if there are any edge cases)

`if (type != SDL_POLLSENTINEL || (include_sentinel && !SDL_AtomicGet(&SDL_sentinel_pending)))`

Reply:
Yep, working on that now.